apps/aurora-portal/src/polaris-bff/trpc.ts

```diff
@@ -1,9 +1,21 @@
-import { initTRPC } from "@trpc/server"
-
+import { initTRPC, TRPCError } from "@trpc/server"
+import { Context } from "./context"
 // You can use any variable name you like.
 // We use t to keep things simple.
-export const t = initTRPC.create()
+export const t = initTRPC.context<Context>().create()
 
 export const router = t.router
-export const publicProcedure = t.procedure
 export const mergeRouters = t.mergeRouters
+
+export const publicProcedure = t.procedure
+
+export const protectedProcedure = publicProcedure.use(function isAuthed(opts) {
```

Comment:
Don't we need types for the opts and return object?

Reply:
we have them! As long as you don’t use intermediary layers in the form of functions, the types will be correctly propagated.